src/test/skript/tests/syntaxes/effects/EffDetonate.sk

```diff
@@ -0,0 +1,22 @@
+test "detonate entity effect":
+	spawn a creeper at event-location
+	assert last spawned entity is valid with "creepers should not detonate when they are first spawned"
+	detonate last spawned entity
+	assert last spawned entity isn't valid with "creepers should instantly detonate when spawned"
+
+test "detonate explosive minecart" when running minecraft "1.19":
+	spawn a minecart with tnt at event-location
+	assert last spawned entity is valid with "minecarts with tnt should not detonate when they are first spawned"
+	detonate last spawned entity
+	assert last spawned entity isn't valid with "minecarts with tnt should instantly detonate when they are first spawned"
+
+test "detonate wind charge" when running minecraft "1.21":
+	spawn a wind charge at event-location
+	assert last spawned entity is valid with "wind charges should not detonate when they are first spawned"
+	detonate last spawned entity
+	assert last spawned entity isn't valid with "wind charges should instantly detonate when spawned"
+
+
+
+# The fireworks test will not be added as fireworks are despawned a tick later and tests are instant
+# The tnt test will not be added as tnts are exploded a tick later and tests are instant
```

Comment:
i think it should be possible by checking if the tnt block is set to air

Reply:
it takes a tick to explode I believe

Reply:
that could be, but the block is set to air before the explosion occurs